core/src/main/scala/kafka/coordinator/group/DelayedJoin.scala

```diff
@@ -57,6 +62,8 @@ private[group] class InitialDelayedJoin(coordinator: GroupCoordinator,
 
   override def tryComplete(): Boolean = false
 
+  override def onExpiration(): Unit = tryToCompleteDelayedAction()
```

Comment:
Hmm, why do we need to override this instead of using the one defined in DelayedJoin?